src/moleculesampleapp/CameraView.java

```diff
@@ -0,0 +1,197 @@
+import javafx.scene.transform.Translate;
+import math.math.geometry.Rotation;
+import math.math.geometry.RotationOrder;
+import math.math.geometry.Vector3D;
+
+/**
+ * Created by Piotr on 04/02/2016.
+ */
+public class CameraView {
+    private double mousePosX;
+    private double mousePosY;
+    private double mouseOldX;
+    private double mouseOldY;
+    private double mouseDeltaX;
+    private double mouseDeltaY;
+
+    private Vec3d angle;
+    private double scale;
+    private Vec3d offset;
+
+    private static final double SCROLL_SCALE = 0.0025;
+    private static final double ROTATE_SCALE = .01;
+    private static final double MOVE_SCALE = 1;
+    private Rotate xRotate, yRotate, zRotate;
+    private Translate xyzTranslate, planeTranslate;
+
+    private PerspectiveCamera camera;
+
+    private Scale cameraScale;
+
+    public Group elements;
+
+    private Group cameraDistance;
+
+    private Group cameraView;
+
+    private SubScene scene;
+
+    CameraView(double width, double height, boolean depthBuffer, SceneAntialiasing sceneAntialiasing) {
+        offset = new Vec3d(0, 0, 0);
+        scale = 0;
+        angle = new Vec3d(0, 0, 0);
+
+        xRotate = new Rotate(0, new Point3D(0, 0, 1));
+        yRotate = new Rotate(0, new Point3D(1, 0, 0));
+        zRotate = new Rotate(0, new Point3D(0, 1, 0));
+
+        xyzTranslate = new Translate(0, 0, 0);
+        planeTranslate = new Translate(width / 2, height / 2, 0);
+
+        cameraScale = new Scale(1, 1, 1);
+
+        elements = new Group();
+
+        elements.getTransforms().addAll(xRotate, yRotate, zRotate, xyzTranslate);
+
+        cameraDistance = new Group(elements);
+
+        cameraDistance.getTransforms().addAll(planeTranslate, cameraScale);
+
+        camera = new PerspectiveCamera(false);
+
+        //camera.getTransforms().add(planeTranslate.createInverse());
+
+        cameraView = new Group(cameraDistance);
+
+        cameraView.getChildren().add(camera);
+
+        scene = new SubScene(cameraView, width, height, depthBuffer, sceneAntialiasing);
+
+        scene.setPickOnBounds(true); //to catch mouse also on "background"
+
+        scene.setCamera(camera);
+
+        setHandlers(scene);
+    }
+
+    public void setHandlers(SubScene subScene) {
+        subScene.setOnMousePressed(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent me) {
+                mousePosX = me.getSceneX();
+                mousePosY = me.getSceneY();
+                mouseOldX = me.getSceneX();
+                mouseOldY = me.getSceneY();
+
+                if (me.isMiddleButtonDown()) {
+
+                    offset.x = offset.y = offset.z = 0;
+                    scale = 0;
+                    angle.x = angle.y = angle.z = 0;
+
+                    xRotate.setAngle(Math.toDegrees(angle.x));
+                    yRotate.setAngle(Math.toDegrees(angle.y));
+                    zRotate.setAngle(Math.toDegrees(angle.z));
```

Comment:
`angle.x = angle.y = angle.z = 0;` so `toDegrees(angle.z)` would be also `0`. Dummy line?